tools:scripts: Added build support for Maxim

CiprianRegus opened the pull request:
Added maxim.mk file and a script to configure VSCode for building
and debugging on Maxim platform.

Signed-off-by: Ciprian Regus <ciprian.regus@analog.com>

CiprianRegus commented:
V2:
- SDK source and header files are now automatically linked.
- Ported the ADXL355 to Maxim platform for testing the build system.

rbolboac commented on projects/eval-adxl355-pmdz/src/main.c:
```diff
@@ -7,38 +7,77 @@
 
+#ifdef STM32_PLATFORM
+	stm32_init();
+	HAL_Init();
+	SystemClock_Config();
```
These lines are not needed anymore. stm32_init is performing these actions instead.

rbolboac commented on projects/eval-adxl355-pmdz/src/main.c:
```diff
@@ -7,38 +7,77 @@
-#define SPI_DEVICE_ID	 1
+#ifdef STM32_PLATFORM
+#define UART_ID			5
+#define SPI_DEVICE_ID	 	1
```
nip: check if indentation is correct

rbolboac commented on projects/eval-adxl355-pmdz/src/main.c:
```diff
@@ -7,38 +7,77 @@
+
+#ifdef MAXIM_PLATFORM
+#define UART_ID			1
+#define SPI_DEVICE_ID		0
```
nip: check if indentation is correct

CiprianRegus commented:
V3:
- Deleted extra tabs in defines.
- Removed HAL_Init() and SystemClock_Config().
- Added path to svd file to debug configuration.
- Running sdkopen updates the current configurations with the TARGET value.

adrimbarean reviewed (commented):
One question and a note. Upon running the build on a windows machine the linker threw the `[object] uses VFP register arguments, [elf] does not` error. This error might be due to my misunderstanding on how to run the build properly so I would suggest having some recepy for running the build written down somewhere. Just `make` didn't work. The command I used was `make PLATFORM=maxim MAXIM_LIBRARIES=C:\MaximSDK\Libraries`
